webapp/src/ts/services/enketo.service.ts

```diff
@@ -355,10 +355,18 @@ export class EnketoService {
         window.history.back();
         this.setupNavButtons($wrapper, form.pages._getCurrentIndex() - 1);
         this.forceRecalculate(form);
+        this.pauseMultimedia($wrapper);
         return false;
       });
   }
 
+  // This code can be removed once this issue is fixed: https://github.com/enketo/enketo-core/issues/816
+  private pauseMultimedia($wrapper) {
+    $wrapper
+      .find('audio, video')
+      .each((idx, element) => element.pause());
```

Comment:
Audio and video media elements are prone to memory leaks  
If we're done playing this media, we should release any reference to it once it's done playing according to [the spec](https://html.spec.whatwg.org/multipage/media.html#best-practices-for-authors-using-media-elements)

Reply:
For Enketo forms, we can't remove it, since users can still click "previous" button and play it again. 
I think this is a constrain to be mindful when designing forms.

Reply:
Sadly, how Enketo works is that it creates all the html pages with all the elements, and makes visible the one that is current. I prefer to not hack it.

Reply:
Ah indeed, we can't simply remove it while the user is still using the form.  
It would be nice if we could hook into a form submit event to release references to audios/videos elements but I'm not sure if that's something we can do with Enketo.

Have we ever had memory leak issues in the webapp in the past?

Reply:
Interesting questions!

> It would be nice if we could hook into a form submit

When the form is submitted, the html is removed and a new Angular partial is displayed (/#/reports/[id]) I don't see we are keeping the media in javascript variables so far (we don't have custom logic _I think_) 
@m5r to double check: If the media elements are gone from HTML, then are we safe of the leak? (I think we do 🤔)

> Have we ever had memory leak issues in the webapp in the past?

Specifically our code in webapp, not really during my time at Medic.
But probably it had happened before. I did a quick search regarding media in core's issues but didn't find anything. 
There is a known issue regarding Enketo where forms get slow when there are complex structures in repeats (groups with inner groups with more groups inside), the advice at that time was to redesign the forms. So probably Enketo lib might have some performance issues.

Reply:
> If the media elements are gone from HTML, then are we safe of the leak?

Not necessarily, that's the tricky thing with HTML media elements :sweat_smile:  
It's recommended to set the element's `src` attribute to an empty string to make sure it's being correctly garbage-collected

Reply:
🤔 That's tricky from the Browser, I wonder if all browser are using the same mechanism. 
I think this can be another ticket, to actually research and evaluate if it's a memory leak and if it's necessary remove the `src`. Very interesting!

Reply:
Of course, your work here is already very valuable and I don't want to block the merge. Let's do that!